app/src/main/java/org/p2p/wallet/home/ui/crypto/MyCryptoPresenter.kt

```diff
@@ -163,31 +167,9 @@ class MyCryptoPresenter(
     }
 
     override fun onClaimClicked(canBeClaimed: Boolean, token: Token.Eth) {
+        analytics.logClaimButtonClicked()
         launch {
-            analytics.logClaimButtonClicked()
-            if (canBeClaimed) {
-                view?.navigateToTokenClaim(token)
-            } else {
-                val latestActiveBundleId = token.latestActiveBundleId ?: return@launch
-                val bridgeBundle = cryptoInteractor.getClaimBundleById(latestActiveBundleId) ?: return@launch
-                val claimDetails = homeMapper.mapToClaimDetails(
-                    bridgeBundle = bridgeBundle,
-                    minAmountForFreeFee = cryptoInteractor.getClaimMinAmountForFreeFee(),
-                )
-                val progressDetails = homeMapper.mapShowProgressForClaim(
-                    amountToClaim = bridgeBundle.resultAmount.amountInToken,
-                    iconUrl = token.iconUrl.orEmpty(),
-                    claimDetails = claimDetails
-                )
-                transactionManager.emitTransactionState(
-                    latestActiveBundleId,
-                    TransactionState.ClaimProgress(latestActiveBundleId)
-                )
-                view?.showProgressDialog(
-                    bundleId = bridgeBundle.bundleId,
-                    progressDetails = progressDetails
-                )
-            }
+            claimHandler.handle(view, canBeClaimed, token)
```

Comment:
I would suggest removing view from the params and return here some action or handle it other way 
But it's just a suggestion